src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,381 @@
+                    ? "none"
+                    : "",
+                height:
+                  this.minorSize / 100 +
+                  this.calcThickness() * (this.minorSize / 150) +
+                  "px",
+                transform:
+                  "translateY(-" +
+                  (this.minorSize / 100 +
+                    this.calcThickness() * (this.minorSize / 150)) +
+                  "px)"
+              }}
+            />
+          </span>
+        ) : (
+          ""
+        )}
+        <div
+          class="gauge"
+          style={{
+            "box-shadow":
+              (this.minorSize <= 300 && this.thickness <= 25) ||
+              !this.styleShadow
+                ? "none"
+                : "",
+            height: `${this.minorSize * 0.792 -
+              this.calcThickness() * (this.minorSize / 100)}px`,
+            width: `${this.minorSize * 0.792 -
+              this.calcThickness() * (this.minorSize / 100)}px`
+          }}
+        >
+          {this.showValue ? (
+            <div
+              style={{
+                "font-size": `${(this.minorSize * 0.795 -
+                  this.calcThickness() / 2 * (this.minorSize / 100)) /
+                  8}px`
+              }}
+            >
+              {this.calcPercentage() > 100
+                ? 100
+                : this.calcPercentage() < 0
+                  ? 0
+                  : `${Math.round(this.calcPercentage())}%`}
+            </div>
+          ) : (
+            ""
+          )}
+        </div>
+      </div>
+    );
+  }
+
+  private renderLine(childRanges) {
+    const divRanges = [];
+    const divRangesName = [];
+    let currentMargin = 0;
+    //////////////////////////////////////////
+    function calcPositionRange(preValue) {
+      return (currentMargin += preValue);
+    }
+
+    function addLineRanges(currentChild, nextChild, component) {
+      return (
+        <div
+          class="range"
+          style={{
+            "background-color": currentChild.color,
+            "box-shadow": !component.styleShadow ? "none" : "",
+            "margin-left": `${calcPositionRange(
+              !!nextChild
+                ? parseInt(nextChild.getAttribute("amount"), 10) *
+                  100 /
+                  component.totValues
+                : 0
+            )}%`,
+            width: `${parseInt(currentChild.getAttribute("amount"), 10) *
+              100 /
+              component.totValues}%`
+          }}
+        />
+      );
+    }
+
+    function addRangeCaption(currentChild, component) {
+      return (
+        <span
+          class="rangeName"
+          style={{
+            "margin-left": `${currentMargin}%`,
+            width: `${parseInt(currentChild.getAttribute("amount"), 10) *
+              100 /
+              component.totValues}%`
+          }}
+        >
+          {currentChild.name}
+        </span>
+      );
+    }
+    //////////////////////////////////////////
```

Comment:
Please reconsider using this type of comments. They don't really add much for other developers that are reading this code.